app/frontend/src/cljs/teet/file/file_controller.cljs

```diff
@@ -118,73 +140,96 @@
                                                 :file/sequence-number
                                                 :file/original-name]))
                      :previous-version-id (:db/id file)}
+           :error-event (partial ->UploadNewVersionError progress-atom)
            :result-event (fn [result]
                            (map->UploadFileUrlReceived
                              (merge result
                                     {:file-data (:file-object new-version)
                                      :on-success (common-controller/->Refresh)})))}))
 
+  UploadNewVersionError
```

Comment:
Was there other difference besides the refresh-fx call in this and FileUploadError? Could maybe same code be used for both if the refresh-fx is harmless here too? I don't know what it does so this is shooting in the dark :)

Reply:
The other one needs to control atom state of the form, to remove the in-progress state.